src/main/java/org/sonarlint/cli/report/ResourceReport.java

```diff
@@ -29,42 +30,37 @@
 import org.sonarlint.cli.util.MutableInt;
 import org.sonarsource.sonarlint.core.IssueListener;
 
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
 import static org.sonarlint.cli.util.Util.getOrCreate;
 
 public final class ResourceReport {
-  private final String filePath;
-  private final boolean project;
+  private final Path filePath;
   private final IssueVariation total = new IssueVariation();
   private final List<IssueListener.Issue> issues = new ArrayList<>();
 
   private final Map<IssueCategory, CategoryReport> reportByCategory = new HashMap<>();
-
   private final Map<Integer, List<IssueListener.Issue>> issuesPerLine = new HashMap<>();
-  private final Map<Integer, List<IssueListener.Issue>> newIssuesPerLine = new HashMap<>();
 
   private final Map<String, MutableInt> issuesByRule = new HashMap<>();
   private final Map<Severity, MutableInt> issuesBySeverity = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Convert this Map to an EnumMap. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1640)